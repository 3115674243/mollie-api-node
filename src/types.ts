import createMollieClient from './createMollieClient';
export default createMollieClient;

export * from './createMollieClient';

export { default as List } from './data/list/List';

export { default as Capture } from './data/payments/captures/Capture';
import {
  GetParameters as CapturesGetParameters,
  ListParameters as CapturesListParameters
} from './resources/payments/captures/parameters';
export { CapturesGetParameters, CapturesListParameters };

export { default as Chargeback } from './data/chargebacks/Chargeback';
import {
  ListParameters as ChargebacksListParameters
} from './resources/chargebacks/parameters';
export { ChargebacksListParameters };

export { default as Customer } from './data/customers/Customer';
import {
  CreateParameters as CustomerCreateParams,
  GetParameters as CustomerGetParams,
  ListParameters as CustomersListParams,
  UpdateParameters as CustomerUpdateParams,
  DeleteParameters as CustomerDeleteParams,
} from './resources/customers/parameters';
export { CustomerCreateParams, CustomerGetParams, CustomersListParams, CustomerUpdateParams, CustomerDeleteParams };

export { default as Mandate } from './data/customers/mandates/Mandate';
import { CreateParameters as MandateCreateParams, GetParameters as MandateGetParams, ListParameters as MandatesListParams, RevokeParameters as MandateRevokeParams } from './resources/customers/mandates/parameters';
export { MandateCreateParams, MandateGetParams, MandatesListParams, MandateRevokeParams };

export { default as Method } from './data/methods/Method';
import { GetParameters as MethodGetParams, ListParameters as MethodsListParams } from './resources/methods/parameters';
export { MethodGetParams, MethodsListParams };

export { default as Order } from './data/orders/Order';
import {
  CreateParameters as OrderCreateParams,
  GetParameters as OrderGetParams,
  ListParameters as OrdersListParams,
  UpdateParameters as OrderUpdateParams,
  CancelParameters as OrderCancelParams,
} from './resources/orders/parameters';
export { OrderCreateParams, OrderGetParams, OrdersListParams, OrderUpdateParams, OrderCancelParams };

export { default as OrderLine } from './data/orders/orderlines/OrderLine';
import { UpdateParameters as OrderLineUpdateParams, CancelParameters as OrderLineCancelParams } from './resources/orders/orderlines/parameters';
export { OrderLineUpdateParams, OrderLineCancelParams };

export { default as Payment } from './data/payments/Payment';
import { CreateParameters as PaymentCreateParams, GetParameters as PaymentGetParams, ListParameters as PaymentsListParams, CancelParameters as PaymentCancelParams } from './resources/payments/parameters';
import { CreateParameters as CustomerPaymentCreateParams, ListParameters as CustomerPaymentsListParams } from './resources/customers/payments/parameters';
import { CreateParameters as OrderPaymentCreateParams } from './resources/payments/orders/parameters';
export {
  PaymentCreateParams,
  PaymentGetParams,
  PaymentsListParams,
  PaymentCancelParams,
  CustomerPaymentCreateParams,
  CustomerPaymentsListParams,
  OrderPaymentCreateParams,
};

export { default as Refund } from './data/refunds/Refund';
import { ListParameters as RefundsListParams } from './resources/refunds/parameters';
import {
  CreateParameters as PaymentRefundCreateParams,
  GetParameters as PaymentRefundGetParams,
  ListParameters as PaymentRefundsListParams,
  CancelParameters as PaymentRefundCancelParams,
} from './resources/payments/refunds/parameters';
export { RefundsListParams, PaymentRefundCreateParams, PaymentRefundGetParams, PaymentRefundsListParams, PaymentRefundCancelParams };

export { default as Shipment } from './data/orders/shipments/Shipment';
import { CreateParameters as ShipmentCreateParams, GetParameters as ShipmentGetParams, ListParameters as ShipmentsListParams, UpdateParameters as ShipmentUpdateParams } from './resources/orders/shipments/parameters';
export { ShipmentCreateParams, ShipmentGetParams, ShipmentsListParams, ShipmentUpdateParams };

export { default as Subscription } from './data/subscription/Subscription';
import {
  CreateParameters as SubscriptionCreateParams,
  GetParameters as SubscriptionGetParams,
  ListParameters as SubscriptionsListParams,
  UpdateParameters as SubscriptionUpdateParams,
  CancelParameters as SubscriptionCancelParams,
} from './resources/customers/subscriptions/parameters';
export { SubscriptionCreateParams, SubscriptionGetParams, SubscriptionsListParams, SubscriptionUpdateParams, SubscriptionCancelParams };

export { CardAudience, CardFailureReason, CardLabel, FeeRegion } from './data/global';
export { Issuer } from './data/Issuer';
export { PaymentInclude } from './data/payments/data';
