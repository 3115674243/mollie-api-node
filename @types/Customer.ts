declare namespace Mollie {
  interface Customer {
    name?: string;
    email?: string;
    locale: Locale;
    metadata?: any;

    // Access token parameters
    testmode?: boolean;
  }
}
