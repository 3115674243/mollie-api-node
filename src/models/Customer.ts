import Model from '../model';
import { ICustomer } from '../types/customer';

/**
 * The `Customer` model
 *
 * {@link ICustomer}
 */
export default class Customer extends Model implements ICustomer {
  public static resourcePrefix = 'cst_';

  public resource = 'customer';
  public id = null;
  public mode = null;
  public name = null;
  public email = null;
  public locale = null;
  public metadata = null;
  public recentlyUsedMethods = null;
  public createdAt = null;
  public _links = {
    self: null,
    documentation: null,
    mandates: null,
    subscriptions: null,
    payments: null,
  };

  // Access token parameters
  public testmode?: boolean;

  /**
   * Customer constructor
   *
   * @public âœ“ This method is part of the public API
   */
  public constructor(props?: Partial<ICustomer>) {
    super();

    Object.assign(this, props);
  }
}
