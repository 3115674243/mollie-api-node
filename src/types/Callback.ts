type Callback<R> = (error: Error | null, result: R) => void;

export default Callback;
