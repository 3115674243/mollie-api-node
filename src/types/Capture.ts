declare namespace Mollie {
  interface ICapture {
    resource: string;
    id: string;
    mode: ApiMode;
    amount: IAmount;
    settlementAmount: IAmount;
    paymentId: string;
    shipmentId?: string;
    settlementId?: string;
    createdAt: string;
    _links: ILinks;
  }

  namespace Params {
    interface Create {
      // Access token parameters
      testmode?: false;
    }
  }
}
