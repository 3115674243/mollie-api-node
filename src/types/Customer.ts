declare namespace Mollie {
  interface ICustomer {
    resource: string;
    id: string;
    mode: ApiMode;
    name: string;
    email: string;
    locale: Locale;
    recentlyUsedMethods: Array<Method>;
    metadata: any;
    createdAt: string;
    _links: ILinks;

    // Access token parameters
    testmode?: boolean;
  }

  namespace Customer {
    namespace Params {
      interface ICreate {
        /** @param name - The full name of the customer.  */
        name?: string;
        /** @param email - The email address of the customer. */
        email?: string;
        /** @param locale - Allows you to preset the language
         *                  to be used in the hosted payment
         *                  pages shown to the consumer.
         *                  When this parameter is not provided,
         *                  the browser language will be used
         *                  instead in the payment flow
         *                  (which is usually more accurate).
         */
        locale?: Locale;
        /** @param metadata - Provide any data you like,
         *                    and we will save the data alongside
         *                    the customer.
         *                    Whenever you fetch the customer with
         *                    our API, weâ€™ll also include the metadata.
         *                    You can use up to 1kB of JSON.
         */
        metadata?: any;

        // Access token parameters
        /** @param testmode - Set this to true to create a test mode customer. */
        testmode?: boolean;
      }

      interface IGet {
        // Access token parameters
        testmode?: boolean;
      }

      interface IUpdate {
        name?: string;
        email?: string;
        locale?: Locale;
        metadata?: any;

        // Access token parameters
        testmode?: boolean;
      }

      interface IDelete {
        // Access token parameters
        testmode?: boolean;
      }

      interface IList {
        from?: string;
        limit?: number;
      }
    }

    namespace Callback {
      type Create = (error: any, customer: ICustomer) => void;
      type Get = (error: any, customer: ICustomer) => void;
      type Update = (error: any, customer: ICustomer) => void;
      type Delete = (error: any, success: boolean) => void;
      type List = (error: any, customers: Mollie.List<ICustomer>) => void;
    }
  }

  namespace CustomerPayments {
    namespace Params {
      interface ICreate {
        customerId: string;

        name?: string;
        email?: string;
        locale: Locale;
        metadata?: any;

        // Access token parameters
        testmode?: boolean;
      }

      interface IList {
        customerId: string;

        from?: string;
        limit?: number;

        // Access token parameters
        testmode?: boolean;
      }
    }

    namespace Callback {
      type Create = (error: any, customer: ICustomer) => void;
      type List = (error: any, customers: Mollie.List<ICustomer>) => void;
    }
  }
}
