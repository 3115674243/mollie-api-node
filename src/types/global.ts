/* eslint-disable @typescript-eslint/camelcase */
export enum Locale {
  en_US = 'en_US',
  nl_NL = 'nl_NL',
  nl_BE = 'nl_BE',
  fr_FR = 'fr_FR',
  fr_BE = 'fr_BE',
  de_DE = 'de_DE',
  de_AT = 'de_AT',
  de_CH = 'de_CH',
  es_ES = 'es_ES',
  ca_ES = 'ca_ES',
  pt_PT = 'pt_PT',
  it_IT = 'it_IT',
  nb_NO = 'nb_NO',
  sv_SE = 'sv_SE',
  fi_FI = 'fi_FI',
  da_DK = 'da_DK',
  is_IS = 'is_IS',
  hu_HU = 'hu_HU',
  pl_PL = 'pl_PL',
  lv_LV = 'lv_LV',
  lt_LT = 'lt_LT',
}

export enum PaymentMethod {
  bancontact = 'bancontact',
  banktransfer = 'banktransfer',
  belfius = 'belfius',
  bitcoin = 'bitcoin',
  creditcard = 'creditcard',
  directdebit = 'directdebit',
  eps = 'eps',
  giftcard = 'giftcard',
  giropay = 'giropay',
  ideal = 'ideal',
  inghomepay = 'inghomepay',
  kbc = 'kbc',
  klarnapaylater = 'klarnapaylater',
  klarnasliceit = 'klarnasliceit',
  paypal = 'paypal',
  paysafecard = 'paysafecard',
  przelewy24 = 'przelewy24',
  sofort = 'sofort',
}

export enum ApiMode {
  test = 'test',
  live = 'live',
}

export interface IImage {
  size1x: string;
  size2x: string;
  svg: string;
}

export interface IUrl {
  href: string;
  type: string;
}

export interface ILinks {
  self: IUrl;
  documentation: IUrl;
}

export interface IListLinks extends ILinks {
  next: IUrl;
  previous: IUrl;
}

export interface IAmount {
  currency: string;
  value: string;
}

export interface IAddress {
  streetAndNumber: string;
  streetAdditional?: string;
  postalCode: string;
  city: string;
  region?: string;
  country: string;
}

export type CardLabel = 'American Express' | 'Carta Si' | 'Carte Bleue' | 'Dankort' | 'Diners' | 'Club' | 'Discover' | 'JCB' | 'Laser' | 'Maestro' | 'Mastercard' | 'Unionpay' | 'Visa' | null;

export type CardFailureReason =
  | 'invalid_card_number'
  | 'invalid_cvv'
  | 'invalid_card_holder_name'
  | 'card_expired'
  | 'invalid_card_type'
  | 'refused_by_issuer'
  | 'insufficient_funds'
  | 'inactive_card';

export type CardAudience = 'consumer' | 'business' | null;

export type FeeRegion = 'intra-eu' | 'other';

export enum SequenceType {
  oneoff = 'oneoff',
  first = 'first',
  recurring = 'recurring',
}

export type IMollieApiErrorLinks = { [key: string]: IUrl } | ILinks;

export interface IMollieApiError {
  status?: number;
  title: string;
  detail: string;
  field?: string;
  _links?: IMollieApiErrorLinks;
}
