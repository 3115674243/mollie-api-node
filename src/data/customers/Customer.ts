import { ApiMode, Links, Url, Locale, PaymentMethod } from '../global';
import Model from '../Model';
import commonHelpers from '../commonHelpers';
import Seal from '../../types/Seal';

/**
 * Customer Response object.
 *
 * @see https://docs.mollie.com/reference/v2/customers-api/get-customer
 */
export interface CustomerData extends Model<'customer'> {
  mode: ApiMode;
  name: string;
  email: string;
  locale: Locale;
  recentlyUsedMethods: PaymentMethod[];
  metadata: Record<string, string>;
  createdAt: string;
  _links: CustomerLinks;
}

type Customer = Seal<CustomerData, typeof commonHelpers>;

export default Customer;

/**
 * Customer _links object
 *
 * @param mandates - Mandates list
 * @param subscriptions - Subscriptions list
 * @param payments - Payments list
 */
export interface CustomerLinks extends Links {
  mandates: Url;
  subscriptions: Url;
  payments: Url;
}

export function injectPrototypes(input: CustomerData): Customer {
  return Object.assign(Object.create(commonHelpers), input);
}
