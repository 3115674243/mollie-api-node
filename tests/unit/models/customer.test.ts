import Customer from '../../../src/models/Customer';
import {ApiMode, Locale, PaymentMethod} from '../../../src/types/global';

describe('customer model', () => {
  it('should instantiate with default values', () => {
    const customer = new Customer();

    expect(customer.toPlainObject()).toMatchSnapshot();
  });

  it('should instantiate with given values', () => {
    const customerProps = {
      resource: 'customer',
      id: 'cst_kEn1PlbGa',
      mode: ApiMode.test,
      name: 'Customer A',
      email: 'customer@example.org',
      locale: Locale.nl_NL,
      metadata: null,
      recentlyUsedMethods: [
        PaymentMethod.creditcard,
        PaymentMethod.ideal,
      ],
      createdAt: '2018-04-06T13:23:21.0Z',
      _links: {
        self: {
          href: 'https://api.mollie.com/v2/customers/cst_kEn1PlbGa',
          type: 'application/hal+json',
        },
        mandates: {
          href: 'https://api.mollie.com/v2/customers/cst_kEn1PlbGa/mandates',
          type: 'application/hal+json',
        },
        subscriptions: {
          href: 'https://api.mollie.com/v2/customers/cst_kEn1PlbGa/subscriptions',
          type: 'application/hal+json',
        },
        payments: {
          href: 'https://api.mollie.com/v2/customers/cst_kEn1PlbGa/payments',
          type: 'application/hal+json',
        },
        documentation: {
          href: 'https://docs.mollie.com/reference/v2/customers-api/get-customer',
          type: 'text/html',
        },
      },
    };
    const customer = new Customer(customerProps);

    expect(customer.toPlainObject()).toMatchSnapshot();
  });
});
