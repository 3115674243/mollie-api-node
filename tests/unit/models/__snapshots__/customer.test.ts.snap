// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`customer model should instantiate with default values 1`] = `
Object {
  "_links": Object {
    "documentation": null,
    "mandates": null,
    "payments": null,
    "self": null,
    "subscriptions": null,
  },
  "createdAt": null,
  "email": null,
  "id": null,
  "locale": null,
  "metadata": null,
  "mode": null,
  "name": null,
  "recentlyUsedMethods": null,
  "resource": "customer",
  "testmode": undefined,
}
`;

exports[`customer model should instantiate with given values 1`] = `
Object {
  "_links": Object {
    "documentation": Object {
      "href": "https://docs.mollie.com/reference/v2/customers-api/get-customer",
      "type": "text/html",
    },
    "mandates": Object {
      "href": "https://api.mollie.com/v2/customers/cst_kEn1PlbGa/mandates",
      "type": "application/hal+json",
    },
    "payments": Object {
      "href": "https://api.mollie.com/v2/customers/cst_kEn1PlbGa/payments",
      "type": "application/hal+json",
    },
    "self": Object {
      "href": "https://api.mollie.com/v2/customers/cst_kEn1PlbGa",
      "type": "application/hal+json",
    },
    "subscriptions": Object {
      "href": "https://api.mollie.com/v2/customers/cst_kEn1PlbGa/subscriptions",
      "type": "application/hal+json",
    },
  },
  "createdAt": "2018-04-06T13:23:21.0Z",
  "email": "customer@example.org",
  "id": "cst_kEn1PlbGa",
  "locale": "nl_NL",
  "metadata": null,
  "mode": "test",
  "name": "Customer A",
  "recentlyUsedMethods": Array [
    "creditcard",
    "ideal",
  ],
  "resource": "customer",
  "testmode": undefined,
}
`;
